typescript/raven.d.ts

```diff
@@ -232,6 +307,81 @@ interface RavenTransportOptions {
     onFailure: () => void;
 }
 
-interface RavenPlugin {
+export interface RavenReportDialogOptions {
+    eventId?: number;
+    dsn?: string;
+    user?: {
+        name?: string;
+        email?: string;
+    };
+}
+
+
+// --- Helper Interfaces for complex Data Structures --------------
+export interface RavenPlugin {
     (raven: RavenStatic, ...args: any[]): RavenStatic;
 }
+
+export interface RavenUserContext {
+    id?: string;
+    username?: string;
+    email?: string;
+    ip_address?: string;
+    extra?: { [prop: string]: any };
+}
+
+export interface RavenBreadcrumb {
+    message: string;
+    data: { [id: string]: string };
+    category: string;
+    level: string;
+    timestamp: number;
+}
+
+export interface RavenOutgoingData {
+    /* either message + stacktrace or exception + culprit */
+    message?: string;
+    stacktrace?: {
+        frames: [RavenStacktraceFrame]
+    };
+    exception?: {
+        values: [{
+            type: string;
+            value: string;
+            stacktrace: {
+                frames: [RavenStacktraceFrame];
+            };
+        }];
+    };
+    culprit?: string;
+    request?: {  /* maybe not included */
+        headers: {
+            'User-Agent': string;
+            Referer: string;
+        };
+        url: string;
+    };
+    breadcrumbs: {
+        values: [RavenBreadcrumb];
+    };
+    user: RavenUserContext;
+    level: string;
+    environment: string;
+    release: string;
+    server_name: string;
+    project: string;
+    logger: string;
+    platform: string;
+    tags: { [id: string]: string };
+    extra: { [prop: string]: any };
+    event_id: string;
+    fingerprint: string;
+}
+
+export interface RavenStacktraceFrame {
+     filename: string;
+     lineno:number;
+     colno: string;
+     function:string;
+     in_app: boolean;
+}
```

Comment:
Yea, so. I would mention that although the length may be comparable to raven.js itself, the complexity is significantly lower; it's static definitions, not procedural code.

That said, it is a maintenance burden, and it's definitely your prerogative whether that's something you want to take up. _Someone_ will have to take it up, if nothing else maintaining definitions in DefinitelyTyped. It could be reduced by making pull request templates and having an issue checkmark for "[x] I updated typings/raven.d.ts if I added to or changed the API". I'd be happy to assist with reviews for those if you'd like.